src/files/pyfingerprint/pyfingerprint.py

```diff
@@ -735,22 +730,29 @@ def downloadImage(self, imageDestination):
             if ( receivedPacketType != FINGERPRINT_DATAPACKET and receivedPacketType != FINGERPRINT_ENDDATAPACKET ):
                 raise Exception('The received packet is no data packet!')
 
-            ## X coordinate of current pixel
-            x = 0
-
-            for i in range(0, len(receivedPacketPayload)):
-
-                ## Thanks to Danylo Esterman <soundcracker@gmail.com> for the "multiple with 17" improvement:
+            imageData.append(receivedPacketPayload)
 
-                ## Draw left 4 Bits one byte of package
-                pixels[x, line] = (receivedPacketPayload[i] >> 4) * 17
-                x = x + 1
+        ## Initialize image
+        resultImage = Image.new('L', (256, 288), 'white')
+        pixels = resultImage.load()
+        row = 0
+        column = 0
 
-                ## Draw right 4 Bits one byte of package
-                pixels[x, line] = (receivedPacketPayload[i] & 0b00001111) * 17
-                x = x + 1
+        for y in range(resultImage.height):
+            for x in range(resultImage.width):
 
-            line = line + 1
+                ## One byte contains two pixels
+                ## Thanks to Danylo Esterman <soundcracker@gmail.com> for the "multiple with 17" improvement:
+                if (x % 2 == 0):
+                    pixels[x, y] = (imageData[row][column]  >> 4) * 17
+                else:
+                    pixels[x, y] = (imageData[row][column] & 0x0F) * 17
+                    column += 1
+
+                    ## Reset
+                    if (column == len(imageData[row])):
```

Comment:
No `colum == len() -1`?

Reply:
It was just a question because this looks a bit like a typical faulty index operation, but may also indented - what do you think @philippmeisberger?

Reply:
If I use `len() -1` exception "list index out of range" occurs, so I say `len()` is intended